components/org.wso2.carbon.identity.conditional.auth.functions.opa/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/opa/InvokeOpaFunctionImpl.java

```diff
@@ -0,0 +1,209 @@
+ */
+package org.wso2.carbon.identity.conditional.auth.functions.opa;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ConnectTimeoutException;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.identity.application.authentication.framework.AsyncProcess;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.JsGraphBuilder;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.*;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.Constants;
+import org.wso2.carbon.identity.conditional.auth.functions.opa.internal.OPAFunctionsServiceHolder;
+import org.wso2.carbon.identity.conditional.auth.functions.opa.util.OPAConstants;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.*;
+
+import static org.apache.http.HttpHeaders.ACCEPT;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+
+/**
+ * Implementation of the {@link InvokeOpaFunction}
+ */
+public class InvokeOpaFunctionImpl implements InvokeOpaFunction {
+
+    private static final Log LOG = LogFactory.getLog(InvokeOpaFunctionImpl.class);
+    private static final String TYPE_APPLICATION_JSON = "application/json";
+
+    private CloseableHttpClient client;
+
+    public CloseableHttpClient getClient() {
+        return client;
+    }
+
+    public void setClient(CloseableHttpClient client) {
+        this.client = client;
+    }
+
+    public InvokeOpaFunctionImpl() {
+
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(5000)
+                .setConnectionRequestTimeout(5000)
+                .setSocketTimeout(5000)
+                .build();
+        client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+    }
+
+    @Override
+    public void invokeOPA(String epUrl, Map<String, Object> payload, Map<String, String> options, Map<String, Object> eventHandlers) {
+
+        JsAuthenticationContext context1 = (JsAuthenticationContext) (payload.get(OPAConstants.CONTEXT));
+        JsStep slot = (JsStep) (((JsSteps) context1.getMember(FrameworkConstants.JSAttributes.JS_STEPS)).getSlot(1));
```

Comment:
check whether ((JsSteps) context1.getMember(FrameworkConstants.JSAttributes.JS_STEPS)) is not null before accessing getSlot